### PR TITLE
Add SIGCHLD Handler in new branch signalHandler.

### PR DESCRIPTION
**Pull request by agauniyal**

### CODE REVIEW

File: handler/sigfunc.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
```
Comment (agauniyal) [inline handler/sigfunc.h:3]: @haris94155 change to `<cstdlib>`

File: handler/sigfunc.h [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+Sigfunc *signal(int signo, Sigfunc *func)
+{
+	struct sigaction act,oact;
+
+	act.sa_handler = func;
+	sigemptyset(&act.sa_mask);
+	act.sa_flags = 0;
+	if(signo == SIGALRM){
+#ifdef SA_INTERRUPT
```
Comment (agauniyal) [inline handler/sigfunc.h:19]: Why are these `#ifdef`s required?

File: handler/sigfunc.h [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+		act.sa_flags |= SA_INTERRUPT;
+#endif
+	} else{
+#ifdef SA_RESTART
```
Comment (agauniyal) [inline handler/sigfunc.h:23]: again here?

File: handler/sigfunc.h [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+	}
+	return(sigfunc);
+}
+
+void sig_chld(int signo)
+{
+	pid_t pid;
+	int stat;
+
+	while( (pid = waitpid(-1, &stat, WNOHANG)) > 0);
```
Comment (agauniyal) [inline handler/sigfunc.h]:
```
prog.cc:47:16: error: use of undeclared identifier 'waitpid'
        while( (pid = waitpid(-1, &stat, WNOHANG)) > 0);
```

via - http://melpon.org/wandbox/permlink/HqfdmbbqdNUrcxDE

**Comment by agauniyal** [comment]
@haris94155 - https://gist.github.com/aspyct/3462238 for reference.

File: handler/sigfunc.h (hunk 1 above)
Comment (haris94155) [inline handler/sigfunc.h]: SA_RESTART is optional flag. When the flag is set, a system call interrupted by this signal(example accept,read,write) will be automatically restarted by the kernel.
If the signal being caught is not SIGALRM, we specify the SA_RESTART flag, if defined.The reason for making a special case for SIGALRM is that the purpose of genearting this signal is normally to place a timeout on an I/O operation, in which case, we want the blocked system call to be interrupted by the signal.

File: handler/sigfunc.h (hunk 2 above)
Comment (tonystaa) [reply handler/sigfunc.h:19]: SA_RESTART is optional flag. When the flag is set, a system call interrupted by this signal(example accept,read,write) will be automatically restarted by the kernel.
If the signal being caught is not SIGALRM, we specify the SA_RESTART flag, if defined.The reason for making a special case for SIGALRM is that the purpose of genearting this signal is normally to place a timeout on an I/O operation, in which case, we want the blocked system call to be interrupted by the signal.

**Pull request merged**